pkg/apis/v1beta1/nodepool_validation.go

```diff
@@ -41,8 +41,9 @@ func (in *NodePool) Validate(_ context.Context) (errs *apis.FieldError) {
 func (in *NodePool) RuntimeValidate() (errs *apis.FieldError) {
 	return errs.Also(
 		in.Spec.Template.Spec.validateTaints(),
-		in.Spec.Template.Spec.validateRequirements(),
-		in.Spec.Template.validateRequirementsNodePoolKeyDoesNotExist(),
+		in.Spec.Template.validateLabels().ViaField("metadata"),
```

Comment:
Did you test these error messages? I don't think that the `ViaField` functions that you are calling here make sense. This is just going to print out `metadata` in the path when you want `spec.template.metadata.<field-name>` to be the path